stdlib/public/Darwin/Foundation/NSStringAPI.swift

```diff
@@ -178,7 +178,10 @@ extension String {
   /// Creates a string by copying the data from a given
   /// C array of UTF8-encoded bytes.
   public init?(utf8String bytes: UnsafePointer<CChar>) {
-    if let str = String(validatingUTF8: bytes) {
+    // This isn't declared to accept nil but in practice it appears some people
+    // manage to pass it nil anyway
+    if let str = String(validatingUTF8: bytes),
+       unsafeBitCast(bytes, to: UInt.self) != 0 {
```

Comment:
We really need to curb some of our excessive unsafe bitcasts here. `UInt(bitPattern: bytes)` should work.